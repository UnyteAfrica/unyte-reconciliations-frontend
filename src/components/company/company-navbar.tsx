import { useContext, useEffect, useState } from "react";
import { NavLink, useLocation } from "react-router-dom";
import { BiChevronDown } from "react-icons/bi";
import { Icon } from "../shared/icon";
import { BrowserComboRoutes, BrowserRoutes } from "@/utils/routes";
import { cx } from "class-variance-authority";
import { twMerge } from "tailwind-merge";
import { clearCredentials, getCompanyInitials } from "@/utils/utils";
import { useQuery, useQueryClient } from "@tanstack/react-query";
import { CompanyQueryKeys } from "@/utils/query-keys";
import { Loader } from "../loader";
import { useLockScroll, useMediaQuery } from "@/utils/hooks";
import { LuMenu } from "react-icons/lu";
import { AiOutlineClose } from "react-icons/ai";
import { getProfile } from "@/services/api/api-base";
import { AuthContext } from "@/context/auth.context";

type UrlLink = {
  text: string;
  url: string;
};

const navLinks: UrlLink[] = [
  {
    text: "Overview",
    url: BrowserComboRoutes.companyDashboard + BrowserRoutes.overview,
  },
  {
    text: "Policies",
    url: BrowserComboRoutes.companyDashboard + BrowserRoutes.policies,
  },
  {
    text: "Claims",
    url: BrowserComboRoutes.pendingCompanyClaims,
  },
  {
    text: "Agents",
    url: BrowserComboRoutes.companyDashboard + BrowserRoutes.agents,
  },
];

export const CompanyNavbar = () => {
  const [isOpen, setIsOpen] = useState(false);
  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
  const location = useLocation();
  const { setIsLoggedIn, setEmail } = useContext(AuthContext);
  const queryClient = useQueryClient();

  const { data: companyDetailsData, isPending: isCompanyDetailsLoading } =
    useQuery({
      queryKey: [CompanyQueryKeys.profile],
      queryFn: () => getProfile(),
    });

  useEffect(() => {
    setIsMobileMenuOpen(false);
    setIsOpen(false);
  }, [location.pathname]);

  const logout = () => {
    clearCredentials();
    setIsLoggedIn(false);
    queryClient.invalidateQueries();
  };

  const toggleDropdown = () => {
    setIsOpen(!isOpen);
  };

  const companyDetails = companyDetailsData?.data;

  useEffect(() => {
    if (companyDetails) setEmail(companyDetails.email);
  }, [companyDetails]);

  const { isMediaQueryMatched } = useMediaQuery(1024);

  useLockScroll(isMobileMenuOpen);

  return (
    <>
      {!isMediaQueryMatched && (
        <div className="relative z-30">
          <header className="px-5 pt-10 pb-4">
            <LuMenu
              onClick={() => setIsMobileMenuOpen(true)}
              size={32}
              className="block"
            />
            <Icon type="logo" className="block" />
          </header>
          <div
            className={twMerge(
              "absolute top-0 left-0 flex w-full transition duration-300 -translate-x-[100%]",
              isMobileMenuOpen && "translate-x-0"
            )}
          >
            <nav className="h-dvh bg-white w-[300px] relative px-5 py-10 flex flex-col justify-between">
              <div className="flex justify-between items-center">
                <Icon type="logo" className="block" />
                <AiOutlineClose
                  onClick={() => setIsMobileMenuOpen(false)}
                  className="right-5 w-6 h-6"
                />
              </div>
              <div className="flex flex-col space-y-8">
                {navLinks.map((navLink, idx) => (
                  <NavLink
                    key={idx}
                    to={navLink.url}
                    className={({ isActive }) =>
                      cx(
                        "font-semibold text-lg duration-300",
                        (isActive ||
                          location.pathname.includes(
                            navLink.text.toLowerCase()
                          )) &&
                          "text-[#25D366] underline underline-offset-[10px] decoration-4"
                      )
                    }
                  >
                    {navLink.text}
                  </NavLink>
                ))}
              </div>
              <div>
                {isCompanyDetailsLoading ? (
                  <Loader />
                ) : (
                  !!companyDetails && (
                    <div id="profile">
                      <div className="flex items-center mb-6">
                        {!!companyDetails.profile_image ? (
                          <img
                            className="h-10 w-10 object-cover inline-block mr-2 rounded-full"
                            src={companyDetails.profile_image}
                            alt=""
                          />
                        ) : (
                          <div className="rounded-full h-10 w-10 bg-gray-200 text-base flex items-center justify-center mr-2">
                            {getCompanyInitials(companyDetails.business_name)}
                          </div>
                        )}

                        <div className="text-base text-[#333] font-medium">
                          {companyDetails.business_name}
                        </div>
                      </div>

                      <button
                        className="block text-[#EB5757] font-medium"
                        onClick={logout}
                      >
                        Sign out
                      </button>
                    </div>
                  )
                )}
              </div>
            </nav>
            <div
              className="grow bg-black/20 h-dvh"
              onClick={() => setIsMobileMenuOpen(false)}
            />
          </div>
        </div>
      )}
      {isMediaQueryMatched && (
        <div
          id="navbar-container"
          className="flex justify-between items-center p-6 px-0 mx-auto max-w-6xl"
        >
          <div>
            <Icon className="shrink-0" type="logo" alt="unyte logo" />
          </div>
          <div className="space-x-24">
            {navLinks.map((navLink, idx) => {
              return (
                <NavLink
                  key={idx}
                  to={navLink.url}
                  className={({ isActive }) =>
                    cx(
                      "font-semibold text-lg duration-300",
                      (isActive ||
                        location.pathname.includes(
                          navLink.text.toLowerCase()
                        )) &&
                        "text-[#25D366] underline underline-offset-[30px] decoration-4"
                    )
                  }
                >
                  {navLink.text}
                </NavLink>
              );
            })}
          </div>

          {isCompanyDetailsLoading ? (
            <Loader />
          ) : (
            !!companyDetails && (
              <div id="profile">
                <div className="relative">
                  <button
                    className="px-4 py-2 text-[#333333] rounded-md"
                    onClick={toggleDropdown}
                  >
                    <div className="space-x-2 flex flex-row items-center">
                      {!!companyDetails.profile_image ? (
                        <img
                          className="rounded-full h-10 w-10 object-cover"
                          src={companyDetails.profile_image}
                          alt=""
                        />
                      ) : (
                        <div className="rounded-full h-10 w-10 p-2 bg-gray-200 text-base flex items-center justify-center">
                          {getCompanyInitials(companyDetails.business_name)}
                        </div>
                      )}
                      <span className="text-lg font-semibold hidden min-[1200px]:inline">
                        {companyDetails.business_name}
                      </span>{" "}
                      <BiChevronDown
                        className={cx(
                          "transition-all duration-300 w-6 h-6",
                          isOpen && "rotate-180"
                        )}
                      />
                    </div>
                  </button>
                  <div
                    className={twMerge(
                      "absolute w-[200px] top-[65px] right-0 mt-2 py-2 z-10 border border-[#e1e1e1] bg-white shadow-lg rounded-lg max-h-0 transition-all duration-300 overflow-hidden opacity-0",
                      isOpen && "max-h-[240px] opacity-100"
                    )}
                  >
                    <div className="px-6 py-4 flex items-center p-2">
                      <span className="text-base text-[#333] font-medium">
                        {companyDetails.business_name}
                      </span>
                    </div>
                    <hr />
                    <NavLink
                      to={
                        BrowserComboRoutes.companyDashboard +
                        BrowserRoutes.profile
                      }
                      className="block px-6 py-4 text-[#333] font-medium"
                    >
                      Profile
                    </NavLink>
                    <hr />
                    <button
                      onClick={logout}
                      className="block px-6 py-4 text-[#EB5757] font-medium"
                    >
                      Sign out
                    </button>
                  </div>
                </div>
              </div>
            )
          )}
        </div>
      )}

      <hr />
    </>
  );
};
