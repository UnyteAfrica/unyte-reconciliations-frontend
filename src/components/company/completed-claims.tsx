import { PageContent } from "@/components/shared/page-content";
import { ClaimsTable } from "../tables/claims-table";
import { Claim, ClaimStatus } from "@/types/types";

export const CompletedClaims = () => {
  const claims: Claim[] = [
    {
      policyNo: "123jkf5402",
      email: "samloco@gmail.com",
      date: "May 7, 2023",
      insurer: "AXA mansard",
      status: ClaimStatus.Completed,
      estimate: 40000,
    },
    {
      policyNo: "123jkf5402",
      email: "samloco@gmail.com",
      date: "May 7, 2023",
      insurer: "AXA mansard",
      status: ClaimStatus.Completed,
      estimate: 40000,
    },
    {
      policyNo: "123jkf5402",
      email: "samloco@gmail.com",
      date: "May 7, 2023",
      insurer: "AXA mansard",
      status: ClaimStatus.Completed,
      estimate: 40000,
    },
    {
      policyNo: "123jkf5402",
      email: "samloco@gmail.com",
      date: "May 7, 2023",
      insurer: "AXA mansard",
      status: ClaimStatus.Completed,
      estimate: 40000,
    },
    {
      policyNo: "123jkf5402",
      email: "samloco@gmail.com",
      date: "May 7, 2023",
      insurer: "AXA mansard",
      status: ClaimStatus.Completed,
      estimate: 40000,
    },
  ];
  return (
    <PageContent
      title="Completed Claims"
      pageTable={<ClaimsTable claims={claims} />}
    />
  );
};
