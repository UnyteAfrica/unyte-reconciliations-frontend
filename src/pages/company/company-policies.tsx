import { CompanyPolicy } from "@/types/types";
import { PageContent } from "@/components/shared/page-content";
import { CompanyPoliciesTable } from "@/components/tables/policy-tables";

const policies: CompanyPolicy[] = [
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    date: "May 7, 2023",
    price: 40000,
    agentId: "ID-023535393",
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    date: "May 7, 2023",
    price: 40000,
    agentId: "ID-023535394",
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    date: "May 7, 2023",
    price: 40000,
    agentId: "ID-023535395",
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    date: "May 7, 2023",
    price: 40000,
    agentId: "ID-023535396",
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    date: "May 7, 2023",
    price: 40000,
    agentId: "ID-023535397",
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    date: "May 7, 2023",
    price: 40000,
    agentId: "ID-023535398",
  },
];

export const CompanyPolicies = () => {
  return (
    <PageContent
      title="Policies"
      pageTable={<CompanyPoliciesTable policies={policies} />}
    />
  );
};
