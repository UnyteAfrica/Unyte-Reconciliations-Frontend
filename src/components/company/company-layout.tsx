import { Outlet } from "react-router-dom";
import { CompanyNavbar } from "./company-navbar";

export const CompanyLayout = () => {
  return (
    <div>
      <CompanyNavbar />
      <Outlet />
    </div>
  );
};
