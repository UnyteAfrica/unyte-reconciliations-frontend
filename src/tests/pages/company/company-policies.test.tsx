import { CompanyPolicies } from "@/pages/company/company-policies";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("Company Policies", () => {
  it("renders correctly", () => {
    const tree = render(
      <BrowserRouter>
        <CompanyPolicies />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
