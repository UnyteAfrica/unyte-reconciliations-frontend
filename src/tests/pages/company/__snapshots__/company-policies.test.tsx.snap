// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Company Policies > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="px-5 pb-10 max-w-[850px] mx-auto"
      >
        <header
          class="mb-8"
        >
          <h1
            class="font-semibold text-2xl mb-6"
          >
            Policies
          </h1>
          <div
            class="relative bg-[#F2F2F2] rounded-lg placeholder:text-[#828282] px-4 py-3 flex items-center mb-6 border-2 border-[#E0E0E0] h-10"
          >
            <img
              alt="search icon"
              class="block mr-2 w-[18px] h-[18px]"
              src="/src/assets/Icons/SearchIcon.svg"
            />
            <input
              class="bg-transparent text-opacity-20 outline-none w-full text-sm font-medium py-3"
              placeholder="Find policy reference"
              type="search"
              value=""
            />
          </div>
          <div
            class="flex w-full"
          >
            <div
              class="border border-[#E0E0E0] p-4 rounded-lg flex flex-col grow rounded-r-none"
            >
              <em
                class="not-italic text-xs text-[#828282] mb-1"
              >
                Total Commission
              </em>
              <em
                class="not-italic font-semibold text-[#333] mb-2"
              >
                â‚¦
                40,000.00
              </em>
              <em
                class="not-italic font-medium text-[10px] text-[#25D366]"
              >
                <svg
                  class="inline-block mr-1 w-3"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 0 0 0 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0 0 13.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
                  />
                </svg>
                Up by 10%
              </em>
            </div>
            <div
              class="border border-[#E0E0E0] p-4 rounded-lg flex flex-col grow rounded-l-none"
            >
              <em
                class="not-italic text-xs text-[#828282] mb-1"
              >
                Total Policies Sold
              </em>
              <em
                class="not-italic font-semibold text-[#333] mb-2"
              >
                50 Policies
              </em>
              <em
                class="not-italic font-medium text-[10px] text-[#25D366]"
              >
                <svg
                  class="inline-block mr-1 w-3"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 1024 1024"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 0 0 0 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0 0 13.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
                  />
                </svg>
                Up by 10%
              </em>
            </div>
          </div>
        </header>
        <main
          class=""
        >
          <div
            class="flex justify-between items-center relative"
          >
            <h2
              class="font-medium"
            >
              Activity
            </h2>
            <button
              class="bg-[#25D366] rounded-full py-1 text-white text-xs w-[62px]"
            >
              <svg
                class="inline-block mr-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 144h448M112 256h288M208 368h96"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
              Filter
            </button>
            <div
              class="h-dvh w-screen bg-black/30 fixed inset-0 transition -z-20 opacity-0 translate-y-[100%]"
            >
              <div
                class="bg-white w-screen absolute bottom-0 rounded-t-2xl transition"
              >
                <div
                  class="p-6"
                >
                  <header
                    class="flex justify-between items-center mb-6"
                  >
                    <h2
                      class="font-semibold text-xl"
                    >
                      Filter Commissions
                    </h2>
                    <button>
                      <img
                        alt=""
                        class="w-6 h-6"
                        src="/src/assets/Icons/close.svg"
                      />
                    </button>
                  </header>
                  <div
                    class="bg-white rounded-lg space-y-6 transition-all"
                    data-testid="filter"
                  >
                    <div
                      class="flex justify-between items-center w-full"
                    >
                      <label
                        class="inline-block mb-0"
                      >
                        Period Range
                      </label>
                      <div
                        class="relative w-full inline-block max-w-[200px]"
                      >
                        <button
                          class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
                        >
                          <div
                            class="space-x-2 flex flex-row items-center justify-between"
                          >
                            <span
                              class="text-base"
                              title="selectBtn"
                            >
                              Daily
                            </span>
                            <svg
                              fill="currentColor"
                              height="1em"
                              stroke="currentColor"
                              stroke-width="0"
                              viewBox="0 0 24 24"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                              />
                            </svg>
                          </div>
                        </button>
                        <div
                          class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
                        >
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Daily
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Weekly
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Monthly
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Yearly
                          </button>
                          <hr />
                        </div>
                      </div>
                    </div>
                    <div
                      class="flex justify-between items-center w-full"
                    >
                      <label
                        class="inline-block mb-0"
                      >
                        Custom Date Range
                      </label>
                      <div
                        class="w-full inline-block max-w-[200px]"
                      >
                        <button
                          class="px-4 py-2 border rounded-md relative font-semibold w-full block mb-1"
                        >
                          November 13, 2023
                          <input
                            class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                            placeholder="date"
                            type="date"
                            value="2023-11-13"
                          />
                        </button>
                        <button
                          class="px-4 py-2 border rounded-md relative font-semibold w-full"
                        >
                          November 13, 2023
                          <input
                            class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                            placeholder="date"
                            type="date"
                            value="2023-11-13"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="flex justify-between items-center w-full"
                    >
                      <label
                        class="inline-block mb-0"
                      >
                        Select Product(s)
                      </label>
                      <div
                        class="relative w-full max-w-[200px]"
                      >
                        <button
                          class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
                        >
                          <div
                            class="space-x-2 flex flex-row items-center justify-between"
                          >
                            <span
                              class="text-base"
                              title="selectBtn"
                            >
                              Comprehensive
                            </span>
                            <svg
                              fill="currentColor"
                              height="1em"
                              stroke="currentColor"
                              stroke-width="0"
                              viewBox="0 0 24 24"
                              width="1em"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                              />
                            </svg>
                          </div>
                        </button>
                        <div
                          class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
                        >
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Comprehensive
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Credit Life
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Card Protect
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Student
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Travel
                          </button>
                          <hr />
                          <button
                            class="w-full font-inter py-4 font-semibold"
                          >
                            Device
                          </button>
                          <hr />
                        </div>
                      </div>
                    </div>
                    <div
                      class="flex justify-between items-center w-full"
                    >
                      <label
                        class="text-[#333] mb-2 block text-base"
                        for="amount"
                      >
                        Amount
                         
                      </label>
                      <input
                        class="block font-sora w-full border rounded-lg outline-none focus:border-black px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold max-w-[200px]"
                        id="amount"
                        placeholder="â‚¦"
                      />
                    </div>
                    <button
                      class="bg-mPrimary text-white w-full rounded-md py-2"
                    >
                      Apply Filter
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="loader mx-auto w-16 h-16 mt-20"
          />
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-5 pb-10 max-w-[850px] mx-auto"
    >
      <header
        class="mb-8"
      >
        <h1
          class="font-semibold text-2xl mb-6"
        >
          Policies
        </h1>
        <div
          class="relative bg-[#F2F2F2] rounded-lg placeholder:text-[#828282] px-4 py-3 flex items-center mb-6 border-2 border-[#E0E0E0] h-10"
        >
          <img
            alt="search icon"
            class="block mr-2 w-[18px] h-[18px]"
            src="/src/assets/Icons/SearchIcon.svg"
          />
          <input
            class="bg-transparent text-opacity-20 outline-none w-full text-sm font-medium py-3"
            placeholder="Find policy reference"
            type="search"
            value=""
          />
        </div>
        <div
          class="flex w-full"
        >
          <div
            class="border border-[#E0E0E0] p-4 rounded-lg flex flex-col grow rounded-r-none"
          >
            <em
              class="not-italic text-xs text-[#828282] mb-1"
            >
              Total Commission
            </em>
            <em
              class="not-italic font-semibold text-[#333] mb-2"
            >
              â‚¦
              40,000.00
            </em>
            <em
              class="not-italic font-medium text-[10px] text-[#25D366]"
            >
              <svg
                class="inline-block mr-1 w-3"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 0 0 0 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0 0 13.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
                />
              </svg>
              Up by 10%
            </em>
          </div>
          <div
            class="border border-[#E0E0E0] p-4 rounded-lg flex flex-col grow rounded-l-none"
          >
            <em
              class="not-italic text-xs text-[#828282] mb-1"
            >
              Total Policies Sold
            </em>
            <em
              class="not-italic font-semibold text-[#333] mb-2"
            >
              50 Policies
            </em>
            <em
              class="not-italic font-medium text-[10px] text-[#25D366]"
            >
              <svg
                class="inline-block mr-1 w-3"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M917 211.1l-199.2 24c-6.6.8-9.4 8.9-4.7 13.6l59.3 59.3-226 226-101.8-101.7c-6.3-6.3-16.4-6.2-22.6 0L100.3 754.1a8.03 8.03 0 0 0 0 11.3l45 45.2c3.1 3.1 8.2 3.1 11.3 0L433.3 534 535 635.7c6.3 6.2 16.4 6.2 22.6 0L829 364.5l59.3 59.3a8.01 8.01 0 0 0 13.6-4.7l24-199.2c.7-5.1-3.7-9.5-8.9-8.8z"
                />
              </svg>
              Up by 10%
            </em>
          </div>
        </div>
      </header>
      <main
        class=""
      >
        <div
          class="flex justify-between items-center relative"
        >
          <h2
            class="font-medium"
          >
            Activity
          </h2>
          <button
            class="bg-[#25D366] rounded-full py-1 text-white text-xs w-[62px]"
          >
            <svg
              class="inline-block mr-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 144h448M112 256h288M208 368h96"
                fill="none"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="32"
              />
            </svg>
            Filter
          </button>
          <div
            class="h-dvh w-screen bg-black/30 fixed inset-0 transition -z-20 opacity-0 translate-y-[100%]"
          >
            <div
              class="bg-white w-screen absolute bottom-0 rounded-t-2xl transition"
            >
              <div
                class="p-6"
              >
                <header
                  class="flex justify-between items-center mb-6"
                >
                  <h2
                    class="font-semibold text-xl"
                  >
                    Filter Commissions
                  </h2>
                  <button>
                    <img
                      alt=""
                      class="w-6 h-6"
                      src="/src/assets/Icons/close.svg"
                    />
                  </button>
                </header>
                <div
                  class="bg-white rounded-lg space-y-6 transition-all"
                  data-testid="filter"
                >
                  <div
                    class="flex justify-between items-center w-full"
                  >
                    <label
                      class="inline-block mb-0"
                    >
                      Period Range
                    </label>
                    <div
                      class="relative w-full inline-block max-w-[200px]"
                    >
                      <button
                        class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
                      >
                        <div
                          class="space-x-2 flex flex-row items-center justify-between"
                        >
                          <span
                            class="text-base"
                            title="selectBtn"
                          >
                            Daily
                          </span>
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                            />
                          </svg>
                        </div>
                      </button>
                      <div
                        class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
                      >
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Daily
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Weekly
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Monthly
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Yearly
                        </button>
                        <hr />
                      </div>
                    </div>
                  </div>
                  <div
                    class="flex justify-between items-center w-full"
                  >
                    <label
                      class="inline-block mb-0"
                    >
                      Custom Date Range
                    </label>
                    <div
                      class="w-full inline-block max-w-[200px]"
                    >
                      <button
                        class="px-4 py-2 border rounded-md relative font-semibold w-full block mb-1"
                      >
                        November 13, 2023
                        <input
                          class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                          placeholder="date"
                          type="date"
                          value="2023-11-13"
                        />
                      </button>
                      <button
                        class="px-4 py-2 border rounded-md relative font-semibold w-full"
                      >
                        November 13, 2023
                        <input
                          class="absolute top-[45px] left-0 invisible opacity-0 h-0 w-0"
                          placeholder="date"
                          type="date"
                          value="2023-11-13"
                        />
                      </button>
                    </div>
                  </div>
                  <div
                    class="flex justify-between items-center w-full"
                  >
                    <label
                      class="inline-block mb-0"
                    >
                      Select Product(s)
                    </label>
                    <div
                      class="relative w-full max-w-[200px]"
                    >
                      <button
                        class="px-4 py-2 text-[#333333] rounded-md border font-semibold w-full"
                      >
                        <div
                          class="space-x-2 flex flex-row items-center justify-between"
                        >
                          <span
                            class="text-base"
                            title="selectBtn"
                          >
                            Comprehensive
                          </span>
                          <svg
                            fill="currentColor"
                            height="1em"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M16.293 9.293 12 13.586 7.707 9.293l-1.414 1.414L12 16.414l5.707-5.707z"
                            />
                          </svg>
                        </div>
                      </button>
                      <div
                        class="absolute z-20 w-full top-[45px] left-0 bg-white border rounded-lg border-[#e5e5e5] max-h-0 transition-all duration-300 overflow-hidden opacity-0"
                      >
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Comprehensive
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Credit Life
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Card Protect
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Student
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Travel
                        </button>
                        <hr />
                        <button
                          class="w-full font-inter py-4 font-semibold"
                        >
                          Device
                        </button>
                        <hr />
                      </div>
                    </div>
                  </div>
                  <div
                    class="flex justify-between items-center w-full"
                  >
                    <label
                      class="text-[#333] mb-2 block text-base"
                      for="amount"
                    >
                      Amount
                       
                    </label>
                    <input
                      class="block font-sora w-full border rounded-lg outline-none focus:border-black px-4 py-2 h-[42px] text-base border-[#e5e7eb] font-semibold max-w-[200px]"
                      id="amount"
                      placeholder="â‚¦"
                    />
                  </div>
                  <button
                    class="bg-mPrimary text-white w-full rounded-md py-2"
                  >
                    Apply Filter
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="loader mx-auto w-16 h-16 mt-20"
        />
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
